src/main/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceService.scala

```diff
@@ -24,20 +26,22 @@ trait WorkspaceService extends HttpService with PerRequestCreator with FireCloud
   lazy val log = LoggerFactory.getLogger(getClass)
   lazy val rawlsWorkspacesRoot = FireCloudConfig.Rawls.workspacesUrl
 
+  def transformSingleWorkspaceRequest(entity: HttpEntity): HttpEntity = {
+    entity.as[RawlsWorkspaceResponse] match {
+      case Right(rwr) => new UIWorkspaceResponse(rwr).toJson.prettyPrint
```

Comment:
With respect to unmarshalling, you're doing the right thing. You could also use spray json to do this, but it's uglier
```
val convertedEntity = entity.toString.parseJson.convertTo[RawlsWorkspaceResponse]
```